### ISSUE TITLE
SA_LOCAL_SELF_COMPUTATION bug

### ISSUE THREAD

**Issue by gloNelson** [spotbugs/spotbugs#1472]
Hi,
I read [SA_LOCAL_SELF_COMPUTATION](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-nonsensical-self-computation-involving-a-variable-e-g-x-x-sa-local-self-computation) on SpotBugs Bug Description page, which mentions that this method performs a nonsensical computation of a local variable with another reference to the same variable (e.g., `x&x` or `x-x`).
So, I wrote test cases as below.
```java
int testSA_LOCAL_SELF_COMPUTATION(){
      int flags = 1;
      return flags ^ flags; // warning SA_LOCAL_SELF_COMPUTATION
  }

  int testSA_LOCAL_SELF_COMPUTATION1(){
      int flags = 1;
      return flags & flags; // no warning (False negative)
  }

  int testSA_LOCAL_SELF_COMPUTATION2(){
      int flags = 1;
      return flags - flags; // no warning (False negative)
  }

  int testSA_LOCAL_SELF_COMPUTATION3(){
      int flags = 1;
      return flags | flags; // no warning (False negative)
  }

```

It can be seen from the results that only `^` will trigger SpotBugs to report **SA_LOCAL_SELF_COMPUTATION**, while other operators will not. 

I tried to search test cases of **SA_LOCAL_SELF_COMPUTATION** in SpotBugs source code and found several test cases of  **SA_FIELD_SELF_COMPUTATION** which detect self computation of **field** instead of **local variables**.
Similar test-cases for **SA_FIELD_SELF_COMPUTATION** executed and all four output **SA_FIELD_SELF_COMPUTATION** errors report.
```java
int x = 1;
  int testSA_FIELD_SELF_COMPUTATION(){
      return x ^ x; // warning SA_FIELD_SELF_COMPUTATION
  }

  int testSA_FIELD_SELF_COMPUTATION1(){
      return x & x; // warning SA_FIELD_SELF_COMPUTATION
  }

  int testSA_FIELD_SELF_COMPUTATION2(){
      return x - x; // warning SA_FIELD_SELF_COMPUTATION
  }

  int testSA_FIELD_SELF_COMPUTATION3(){
      return x | x; // warning SA_FIELD_SELF_COMPUTATION
  }
```

Finally, I realized that the fault is in line **325** of [FindSelfComparison.java](https://github.com/spotbugs/spotbugs/blob/a6f9acb2932b54f5b70ea8bc206afb552321a222/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindSelfComparison.java#L325).
```java
else if (opCode == Const.IXOR && item0.sameValue(item1)) {
```
The code shows that SpotBugs reports **SA_FIELD_SELF_COMPUTATION** only when `opCode` equals `Const.IXOR`(`^`).

It is so confusing that self computation detection of `local variables` only including `^` operator while the detection of `field` supports much more. Is it a bug?

**Comment by gloNelson** [comment]
Another problematic code section in [FindSelfComparison.java](https://github.com/spotbugs/spotbugs/blob/a6f9acb2932b54f5b70ea8bc206afb552321a222/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindSelfComparison.java) is mentioned in #1473 (False negatives on SA_FIELD_SELF_COMPUTATION and SA_LOCAL_SELF_COMPUTATION) in detail.

Thanks a lot. @KengoTODA

**Comment by gloNelson** [comment]
I have created a pull request to fix this. Please help me review this :) @KengoTODA Thks!

**Comment by gtoison** [comment]
Fixed by #1478, thank you!
